Replace 'dict_label' argument by 'path'.

mschmidt87 opened the pull request:
- Introduce new keyword argument `path` in save function
- I set the default value for `path` to `None` instead of `''` because I find this clearer.
- Add DeprecationWarning if `dict_label` is used.
- Add Exception if `dict_label` and `path` are simultaneously used
- Adapt tests for new argument name and add test for exception

Fixes #60 .

jakobj commented on h5py_wrapper/wrapper.py:
```diff
@@ -83,6 +83,15 @@ def save(filename, d, write_mode='a', overwrite_dataset=False,
+                                     "be defined simultaneously.")
+                path = dict_label                
+            if path:
                 base = f.require_group(dict_label)
```
shouldn't you use `path` here instead of `dict_label`?

jakobj commented on h5py_wrapper/wrapper.py:
```diff
@@ -83,6 +83,15 @@ def save(filename, d, write_mode='a', overwrite_dataset=False,
         try:
             if dict_label:
+                warnings.warn("Deprecated argument dict_label provided. "
+                              "`dict_label` will be removed in the next release. "
```
we should be consistent with use of the backtick \`, either we use it around every variable name or around none.

mschmidt87 commented:
Thanks for raising the two points. I fixed the bug in `require_path` and removed the backticks since we haven't used them in any other message. Though I would actually prefer using them for argument and function names. But I don't know the general convention.

mschmidt87 commented:
P.S.: No idea about travis. See #62 .
